src/veins/modules/mobility/traci/TraCICommandInterface.cc

```diff
@@ -440,6 +455,28 @@ double TraCICommandInterface::Vehicle::getAccumulatedWaitingTime() const
     return traci->genericGetDouble(CMD_GET_VEHICLE_VARIABLE, nodeId, VAR_WAITING_TIME_ACCUMULATED, RESPONSE_GET_VEHICLE_VARIABLE);
 }
 
+uint8_t TraCICommandInterface::Vehicle::getStopState() const
+{
+    return traci->genericGetInt(CMD_GET_VEHICLE_VARIABLE, nodeId, VAR_STOPSTATE, RESPONSE_GET_VEHICLE_VARIABLE);
+}
+
+bool TraCICommandInterface::Vehicle::isStopped() const
+{
+    return getStopState() & 0x1;
```

Comment:
this reads like it will only catch some reasons because of which a vehicle might be stopped, which would make the method name somewhat of a misnomer. Can we do entirely without it?

Reply:
As far as I'm understanding it, this bit should be set in every case where the vehicle is stopped. I basically adopted it from the Python TraCI library (here: [https://github.com/eclipse/sumo/blob/556418361e6bb34aba810e893f72a017bfed3522/tools/traci/_vehicle.py#L903](url)) since it's easier to use than plain `getStopState()`. I opted not to port the other associated helper methods since I haven't worked with these features of SUMO to test them.

The corresponding SUMO code (https://github.com/eclipse/sumo/blob/c3c72bd2084c8e8f0e30523887253eacdec6d610/src/libsumo/Vehicle.cpp#L493) sets this bit when the vehicle is stopped (`MSVehicle::isStopped()` is true) and the next scheduled stop has reached. `MSVehicle::isStopped()` is implemented by checking whether there is a next stop and it has been reached (https://github.com/eclipse/sumo/blob/ab394a0d9bbf7a2cb86af747dbd982becd1bff00/src/microsim/MSBaseVehicle.cpp#L817). So I believe this bit should always be set when `getStopState() != 0` for any kind of stop.

So as far as I understand SUMO, it should return the expected result for all kinds of planned stops, but not, for example, when the vehicle is waiting at an intersection (technically also "stopped"). If you think this may lead to confusion, or is not necessary, I can remove it.

Reply:
Thanks for the detailed reasoning. I can see that SUMO itself refers to the flag as just "stopped" and its official Python API offers a "isStopped" method to check it. Still, I would fear that offering this method under a name of `isStopped` has a high potential for confusing users. How would you feel about calling the method `isStopReached` (that is, following [the SUMO source code](https://github.com/eclipse/sumo/blob/c3c72bd2084c8e8f0e30523887253eacdec6d610/src/libsumo/Vehicle.cpp#L503) instead of its documentation)?

Reply:
That seems like a sensible approach, I renamed the method accordingly.